traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -0,0 +1,1959 @@
+	my $log_object_rolling_interval_sec = $data->{"LogObject.RollingIntervalSec"} || "";
+	my $log_object_rolling_offset_hr    = $data->{"LogObject.RollingOffsetHr"}    || "";
+	my $log_object_rolling_size_mb      = $data->{"LogObject.RollingSizeMb"}      || "";
+	my $format                          = $data->{"LogFormat.Format"};
+	$format =~ s/"/\\\"/g;
+	$text .= "<LogFormat>\n";
+	$text .= "  <Name = \"" . $log_format_name . "\"/>\n";
+	$text .= "  <Format = \"" . $format . "\"/>\n";
+	$text .= "</LogFormat>\n";
+	$text .= "<LogObject>\n";
+	$text .= "  <Format = \"" . $log_object_format . "\"/>\n";
+	$text .= "  <Filename = \"" . $log_object_filename . "\"/>\n";
+	$text .= "  <RollingEnabled = " . $log_object_rolling_enabled . "/>\n" unless defined();
+	$text .= "  <RollingIntervalSec = " . $log_object_rolling_interval_sec . "/>\n";
+	$text .= "  <RollingOffsetHr = " . $log_object_rolling_offset_hr . "/>\n";
+	$text .= "  <RollingSizeMb = " . $log_object_rolling_size_mb . "/>\n";
+	$text .= "</LogObject>\n";
+
+	return $text;
+}
+
+sub storage_dot_config_volume_text {
+	my $prefix  = shift;
+	my $letters = shift;
+	my $volume  = shift;
+
+	my $text = "";
+	my @postfix = split( /,/, $letters );
+	foreach my $l ( sort @postfix ) {
+		$text .= $prefix . $l;
+		$text .= " volume=" . $volume;
+		$text .= "\n";
+	}
+	return $text;
+}
+
+sub storage_dot_config {
+	my $self        = shift;
+	my $profile_obj = shift;
+	my $filename    = shift;
+	my $scope       = shift;
+
+	my $text = $self->header_comment( $profile_obj->name );
+	my $data = $self->profile_param_data( $profile_obj->id, "storage.config" );
+
+	my $next_volume = 1;
+	if ( defined( $data->{Drive_Prefix} ) ) {
+		$text .= storage_dot_config_volume_text( $data->{Drive_Prefix}, $data->{Drive_Letters}, $next_volume );
+		$next_volume++;
+	}
+
+	if ( defined( $data->{RAM_Drive_Prefix} ) ) {
+		$text .= storage_dot_config_volume_text( $data->{RAM_Drive_Prefix}, $data->{RAM_Drive_Letters}, $next_volume );
+		$next_volume++;
+	}
+
+	if ( defined( $data->{SSD_Drive_Prefix} ) ) {
+		$text .= storage_dot_config_volume_text( $data->{SSD_Drive_Prefix}, $data->{SSD_Drive_Letters}, $next_volume );
+		$next_volume++;
+	}
+
+	return $text;
+}
+
+sub to_ext_dot_config {
+	my $self       = shift;
+	my $server_obj = shift;
+	my $filename   = shift;
+	my $scope      = shift;
+
+	my $text = $self->header_comment( $server_obj->host_name );
+
+	# get the subroutine name for this file from the parameter
+	my $subroutine = $self->profile_param_value( $server_obj->profile->id, $filename, 'SubRoutine', undef );
+	$self->app->log->error( "ToExtDotConfigFile == " . $subroutine );
+
+	if ( defined $subroutine ) {
+		my $package;
+		( $package = $subroutine ) =~ s/(.*)(::)(.*)/$1/;
+		eval "use $package;";
+
+		# And call it - the below calls the subroutine in the var $subroutine.
+		no strict 'refs';
+		$text .= $subroutine->( $self, $server_obj->host_name, $filename );
+		return $text;
+	}
+	else {
+		return;
```

Comment:
do you want to return nothing? or do you want to return $text which at this point only contains the header_comment? if nothing, maybe kill the "else" and just add return undef to be more explicit. or just kill the else entirely.